File: Problems/14-Search-a-2D-Matrix/Search-a-2D-Matrix.cpp
class Solution {
public:
    int getValue(vector<vector<int>>& matrix, int mid)
    {
        int n = matrix.size();
        int m = matrix[0].size();
        int i = mid / m;
        int j = mid % m;
        return matrix[i][j];
    }

    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int n = matrix.size();
        int m = matrix[0].size();
        int left = 0, right = n * m-1;
        while (left <= right)
        {
            int mid = (right - left)/2+left;
            int value = getValue(matrix, mid);
            if (target == value) return true;
            if (target < value) right = mid-1;
            else left = mid + 1;
        }
        return false;
    }
};

File: Problems/14-Search-a-2D-Matrix/search_a_2D_Matrix.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        k = 0
        
        for i in range(len(matrix)):
            if(target ==  matrix[i][0]): return True
            if(target>  matrix[i][0]): k = i   
        for j in range(1,len(matrix[k])):
            if(target == matrix[k][j]): return True
        return False